agent/tool-scripts/pprof

```diff
@@ -101,88 +102,75 @@ done
+            oc get --raw /debug/pprof/profile?seconds=${interval} --server https://${master}:8444 > ${master}_controller_profile && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_controller_profile.txt /tmp/${master}_controller_profile &
+            master_pids[${master}_controller]=$!
+            oc get --raw /debug/pprof/heap --server https://${master}:8444 > ${master}_controller_heap && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_controller_heap.txt /tmp/${master}_controller_heap &
+            if (( i++%${#masters[@]}==0 )); then
+		for pid in ${master_pids[*]}; do
+                    wait ${pid}
+                done
+            fi
+        done
+    done
 }
 
-collect_data() {
+get_hosts() {
+    declare -A node_pids
     while true; do
-        if [ $(systemctl list-unit-files | grep atomic-openshift-master.service | awk {'print $2'}) == "masked" ]; then
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_controller.txt http://localhost:6060/debug/pprof/profile 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_controller_heap.txt http://localhost:6060/debug/pprof/heap 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_api.txt http://localhost:6061/debug/pprof/profile
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_api_heap.txt http://localhost:6061/debug/pprof/heap 
-        else
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S).txt http://localhost:6060/debug/pprof/profile 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_heap.txt http://localhost:6060/debug/pprof/heap 
-        fi
-    done 
-} 
+        j=1
+        for host in "${hosts[@]}"; do
+            oc get --raw /debug/pprof/profile?seconds=${interval} --server https://${host}:10250 > ${host}_profile && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${host}/$(date +%Y-%m-%d:%H:%M:%S)_profile.txt /tmp/${host}_profile &
+            node_pids[${host}]=$!
+            oc get --raw /debug/pprof/heap --server https://${host}:10250 > ${host}_heap && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${host}/$(date +%Y-%m-%d:%H:%M:%S)_heap.txt /tmp/${host}_heap &
+            if (( j++%${#hosts[@]}==0 )); then
+		for pid in ${node_pids[*]}; do
+                    wait ${pid}
+                done
+            fi
+        done
+    done
+}
+
+## check if the inventory variable is set correctly
```

Comment:
Can we move this check to immediately following the argument processing while loop/case statement?

Reply:
Moved in https://github.com/distributed-system-analysis/pbench/pull/847

Reply:
Because it would seem that we don't want to allow an error condition to create pids without reaping them, filling a system.  Making sure we `wait` for all pids we fork prevents that from happening.

Reply:
I think this reply is for the pids?

Reply:
yes, not sure what is happening with github.

I posted that reply with the pids, and now it ends up here.